x-pack/legacy/plugins/apm/public/components/app/Settings/AgentConfigurations/AddEditFlyout/index.tsx

```diff
@@ -0,0 +1,279 @@
+    (
+      idx(selectedConfig, _ => _.settings.transaction_max_spans) || ''
+    ).toString()
+  );
+
+  const isSampleRateValid = isRight(transactionSampleRateRt.decode(sampleRate));
+  const isTransactionMaxSpansValid = isRight(
+    transactionMaxSpansRt.decode(transactionMaxSpans)
+  );
+  const isFormValid =
+    !!serviceName &&
+    !!environment &&
+    isSampleRateValid &&
+    !!captureBody &&
+    isTransactionMaxSpansValid;
+
+  const handleSubmitEvent = async (
+    event:
+      | React.FormEvent<HTMLFormElement>
+      | React.MouseEvent<HTMLButtonElement>
+  ) => {
+    event.preventDefault();
+    setIsSaving(true);
+
+    if (sampleRate === '' || captureBody === '' || transactionMaxSpans === '') {
+      throw new Error('Missing arguments');
+    }
+
+    await saveConfig({
+      serviceName,
+      environment,
+      sampleRate,
+      captureBody,
+      transactionMaxSpans,
+      configurationId: selectedConfig ? selectedConfig.id : undefined
+    });
+    setIsSaving(false);
+    onSaved();
+  };
+
+  return (
+    <EuiPortal>
+      <EuiFlyout size="s" onClose={onClose} ownFocus={true}>
+        <EuiFlyoutHeader hasBorder>
+          <EuiTitle>
+            {selectedConfig ? (
+              <h2>{t('editConfigTitle', 'Edit configuration')}</h2>
+            ) : (
+              <h2>{t('createConfigTitle', 'Create configuration')}</h2>
+            )}
+          </EuiTitle>
+        </EuiFlyoutHeader>
+        <EuiFlyoutBody>
+          <EuiText size="s">
+            This allows you to fine-tune your agent configuration directly in
+            Kibana. Best of all, changes are automatically propagated to your
+            APM agents so there’s no need to redeploy.
+          </EuiText>
+
+          <EuiSpacer size="m" />
+
+          <EuiForm>
+            {/* eslint-disable-next-line jsx-a11y/no-noninteractive-element-interactions */}
+            <form
+              onKeyPress={e => {
+                const didClickEnter = e.which === 13;
+                if (didClickEnter) {
+                  handleSubmitEvent(e);
+                }
+              }}
+            >
+              <ServiceSection
+                selectedConfig={selectedConfig}
+                environment={environment}
+                setEnvironment={setEnvironment}
+                serviceName={serviceName}
+                setServiceName={setServiceName}
+              />
+
+              <EuiSpacer />
+
+              <SettingsSection
+                sampleRate={{
+                  value: sampleRate,
+                  set: setSampleRate,
+                  isValid: isSampleRateValid
+                }}
+                captureBody={{
+                  value: captureBody,
+                  set: setCaptureBody
+                }}
+                transactionMaxSpans={{
+                  value: transactionMaxSpans,
+                  set: setTransactionMaxSpans,
+                  isValid: isTransactionMaxSpansValid
+                }}
```

Comment:
I'm not sure it is the best approach to group the values like this. I could also drop the grouping:
```tsx
<SettingsSection
  // sample rate
  sampleRate={sampleRate}
  setSampleRate={setSampleRate}
  isSampleRateValid={isSampleRateValid}

  // capture body  
  captureBody={captureBody}
  setCaptureBody={setCaptureBody}

  // transaction max spans
  transactionMaxSpans={transactionMaxSpans}
  setTransactionMaxSpans={setTransactionMaxSpans}
  isTransactionMaxSpansValid={isTransactionMaxSpansValid}
/>;
```